src/box/xlog.c

```diff
@@ -730,30 +708,9 @@ xdir_collect_inprogress(struct xdir *xdir)
 		char *ext = strrchr(dent->d_name, '.');
 		if (ext == NULL || strcmp(ext, inprogress_suffix) != 0)
 			continue;
-
-		char path[PATH_MAX];
-		int rc;
-		if ((size_t)snprintf(path, sizeof(path), "%s/%s", dirname,
-				     dent->d_name) >= sizeof(path)) {
-			/*
-			 * If we allocated a large enough buffer for the path,
-			 * unlink would fail with this error according to the
-			 * POSIX standard.
-			 */
-			errno = ENAMETOOLONG;
-			rc = -1;
-		} else {
-			rc = unlink(path);
-		}
-		if (rc < 0)
-			/*
-			 * In case the path got truncated, print the expected
-			 * path.
-			 */
-			say_syserror("error while removing %s/%s",
-				     dirname, dent->d_name);
-		else
-			say_info("removed %s", path);
+		const char *filename = tt_snprintf(PATH_MAX, "%s/%s",
```

Comment:
Yes, this isn't entirely correct, but:
- The worst that can happen because of this is that we don't delete an incomplete file (a file with the `.inprogress` suffix). We can't erroneously delete a materialized file since there can't be two files with the same signature so that one is materialized while the other is incomplete.
- We already use `tt_snprintf(PATH_MAX)` in `xdir_format_filename` so we have exactly the same problem in `xdir_collect_garbage`.

I think that if we decide that this is a problem, we should solve it in all places in a separate patch. For now, I suggest leaving this as is for simplicity and because nobody's ever complained (it's unlikely that anybody uses such long paths).